### PR TITLE
clientkit: add Authentication API

### PR DESCRIPTION
**Pull request by webispy**
add authentication API to support OAuth2 discovery, client_credentials
and token refresh feature.

Signed-off-by: Inho Oh <webispy@gmail.com>

### CODE REVIEW

File: include/clientkit/nugu_auth.hh [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+/**
+ * @brief GrantType
+ */
+enum GrantType {
+    AUTHORIZATION_CODE = 0x1, /* OAuth2 Authorization code */
+    CLIENT_CREDENTIALS = 0x2, /* OAuth2 Client credentials */
+    REFRESH_TOKEN = 0x4, /* OAuth2 refresh token */
+    DEVICE_CODE = 0x8 /* OAuth2 device code */
+};
+
```
Comment (kimhyungrok) [inline include/clientkit/nugu_auth.hh:46]: How about to use enum class if not exist specific reason? That's more robust.

File: include/clientkit/nugu_auth.hh [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+ */
+struct NuguDeviceConfig {
+    std::string OAuthServerUrl; /* OAuth2 server url */
+    std::string OAuthClientId; /* Client id */
+    std::string OAuthClientSecret; /* Client secret */
+    std::string OAuthRedirectUri; /* Redirect uri */
+    std::string DeviceTypeCode; /* Device type code */
+    std::string PoCId; /* PoC id */
+};
+
```
Comment (kimhyungrok) [inline include/clientkit/nugu_auth.hh:58]: Please check variable naming convention on the whole. It mix the camelCase and underscore way.

Comment (webispy) [reply include/clientkit/nugu_auth.hh:58]: Thank you! All variable names have been changed according to coding conventions.

File: include/clientkit/nugu_auth.hh (hunk 1 above)
Comment (webispy) [reply include/clientkit/nugu_auth.hh:46]: Thank you! It seems good to use enums as-is, as there are advantages to bitwise operations.

**Review by kimhyungrok — APPROVED** [review]

**Pull request merged**